runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp

```diff
@@ -48,18 +50,41 @@ class GC_ConstantPoolObjectSlotIterator
 	U_32 *_cpDescriptionSlots;
 	U_32 _cpDescription;
 	UDATA _cpDescriptionIndex;
+	bool _condyOnly;
+	bool _condyEnabled;
```

Comment:
I'm not strictly opposed to this _condyEnabled flag, but wondring if it's really necessary. Would not iterator even without this flag return null in a case the system is not 'enabled' by simply not finding any slot of Constant Dynamic type?

Reply:
If it's really important, I suspect a compile-time test of the JDK level could be used. Recent discussions with @charliegracie implied that we may want to scan Strings as well (stop tenuring String literals immediately).

Reply:
I guess, there is performance implication, since the flag would help avoid the search, but then, I'm not sure how often we would instantiate and use the iterator with condyOnly on a not enabled system?